File: DesktopSample/API/Section/CrossSectionControls/VB.NET/CrossSectionControls/ViewerForm.vb
Imports System.Text
Imports System.Xml
Imports GrapeCity.ActiveReports

Public Class ViewerForm

	Enum ReportType
		CrossSectionControls
		RepeatToFill
		PrintAtBottom
	End Enum

	Sub New()
		Encoding.RegisterProvider(CodePagesEncodingProvider.Instance)
		InitializeComponent()
	End Sub

	Private Delegate Sub LoadReportInvoker(ByVal reportType As ReportType)

	Private Sub ViewerForm_Load(ByVal sender As System.Object, ByVal e As EventArgs) Handles MyBase.Load
		' レポートが描画中に、フォームを開きます。
		BeginInvoke(New LoadReportInvoker(AddressOf LoadReport), ReportType.CrossSectionControls)
		BeginInvoke(New LoadReportInvoker(AddressOf LoadReport), ReportType.RepeatToFill)
		BeginInvoke(New LoadReportInvoker(AddressOf LoadReport), ReportType.PrintAtBottom)
	End Sub

	Private Sub LoadReport(ByVal rptType As ReportType)
		' 新しいInvoiceレポートをインスタンス化します。
		Dim report As New SectionReport
		report.LoadLayout(XmlReader.Create(My.Resources.Invoice))
		report.MaxPages = 10

		' 接続文字列にサンプルデータベースを参照します。
		CType(report.DataSource, Data.DbDataSource).ConnectionString = My.Resources.ConnectionString
		Select Case rptType
			Case ReportType.CrossSectionControls
				cscViewer.LoadDocument(report)
			Case ReportType.RepeatToFill
				CType(report.Sections("Detail"), SectionReportModel.Detail).RepeatToFill = True
				repeatToFillViewer.LoadDocument(report)
			Case ReportType.PrintAtBottom
				CType(report.Sections("customerGroupFooter"), SectionReportModel.GroupFooter).PrintAtBottom = True
				printAtBottomViewer.LoadDocument(report)
		End Select
	End Sub

End Class

File: DesktopSample/API/Section/CrossSectionControls/C#/CrossSectionControls/ViewerForm.cs
using System;
using System.Windows.Forms;
using System.Xml;
using GrapeCity.ActiveReports;
using GrapeCity.ActiveReports.Data;
using GrapeCity.ActiveReports.SectionReportModel;

namespace ActiveReports.Samples.CrossSectionControls
{
	public partial class ViewerForm : Form
	{
		enum ReportType
		{
			CrossSectionControls,
			RepeatToFill,
			PrintAtBottom
		}

		private delegate void LoadReportInvoker(ReportType reportType);

		public ViewerForm()
		{
			InitializeComponent();
		}

		private void ViewerForm_Load(object sender, EventArgs e)
		{
			// レポートが描画中に、フォームを開きます。
			BeginInvoke(new LoadReportInvoker(LoadReport), ReportType.CrossSectionControls);
			BeginInvoke(new LoadReportInvoker(LoadReport), ReportType.PrintAtBottom);
			BeginInvoke(new LoadReportInvoker(LoadReport), ReportType.RepeatToFill);
		}

		void LoadReport(ReportType reportType)
		{
			// 新しいInvoiceレポートをインスタンス化します。
			var report = new SectionReport();
			report.LoadLayout(XmlReader.Create(Properties.Resources.Invoice));

			report.MaxPages = 10;

			// 接続文字列にサンプルデータベースを参照します。
			((DbDataSource)report.DataSource).ConnectionString = Properties.Resources.ConnectionString;			
			switch (reportType)
			{
				case ReportType.CrossSectionControls:
					cscViewer.LoadDocument(report);
					break;
				case ReportType.RepeatToFill:
					((Detail) report.Sections["Detail"]).RepeatToFill = true;
					repeatToFillViewer.LoadDocument(report);
					break;
				case ReportType.PrintAtBottom:
					((GroupFooter) report.Sections["customerGroupFooter"]).PrintAtBottom = true;
					printAtBottomViewer.LoadDocument(report);
					break;
			}
		}
	}
}
